File: TP4/incompleto/maior.pl
maior(XS,M) :- XS = [X|_], encontraMaior(XS, X, M).

encontraMaior([], A, A).
encontraMaior([X|XS], A, M) :- X > A, encontraMaior(XS, X, M).
encontraMaior([X|XS], A, M) :- X =< A, encontraMaior(XS, A, M).

File: TP3/Not in TP/maior.hs
maior :: [Int] -> Int
maior[] = 0
maior (x:xs) = maiorAux x xs


maiorAux :: Int -> [Int] -> Int
maiorAux m [] = m
maiorAux m (x:xs)
   | x > m = maiorAux x xs
   | otherwise = maiorAux m xs